.github/workflows/test-integration.yml

```diff
@@ -28,38 +67,191 @@ jobs:
+
+      - name: Checkout acvm sources
+        uses: actions/checkout@v3 # v3 is needed here otherwise this fails in local execution
+        with:
+          repository: noir-lang/acvm
+          path: acvm
+
+      - name: Download nargo binary
+        uses: actions/download-artifact@v3
+        with:
+          name: nargo
+          path: ./nargo
+
+      - name: Download wasm package artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: noir_wasm
+          path: ./compiler/wasm
+
+      - name: Download noirc package artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: noirc_abi_wasm
+          path: ./tooling/noirc_abi_wasm
+
+      - name: Set nargo on PATH
+        run: |
+          nargo_binary="${{ github.workspace }}/nargo/nargo"
+          chmod +x $nargo_binary
+          echo "$(dirname $nargo_binary)" >> $GITHUB_PATH
+          export PATH="$PATH:$(dirname $nargo_binary)"
+          nargo -V
 
       - name: Install Yarn dependencies
         uses: ./.github/actions/setup
 
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Run Anvil
+        run: |
+          anvil > /dev/null 2>&1 &
+          sleep 10
+
+      - name: Create new Foundry project
+        run: forge init --no-git --no-commit --force foundry-project
+
+      - name: Run codegen-verifier for 1_mul
+        working-directory: tooling/nargo_cli/tests/execution_success/1_mul
+        run: nargo codegen-verifier
+
+      - name: Run codegen-verifier for double_verify_proof
+        working-directory: tooling/nargo_cli/tests/execution_success/double_verify_proof
+        run: nargo codegen-verifier
+
+      - name: Copy compiled contracts
+        run: |
+          cp tooling/nargo_cli/tests/execution_success/1_mul/contract/1_mul/plonk_vk.sol foundry-project/src/1_mul.sol
+          cp tooling/nargo_cli/tests/execution_success/double_verify_proof/contract/double_verify_proof/plonk_vk.sol foundry-project/src/double_verify_proof.sol
+
+      - name: Forge build
+        working-directory: foundry-project
+        run: forge build
+
+      - name: Forge deploy
+        working-directory: foundry-project
+        run: |
+          mul_result=$(forge create --rpc-url http://127.0.0.1:8545 --mnemonic "test test test test test test test test test test test junk" src/1_mul.sol:UltraVerifier)
+          echo "$mul_result"
+          address=$(echo "$mul_result" | grep "Deployed to:" | awk '{print $3}')
+          echo "MUL_CONTRACT_ADDRESS=$address" >> $GITHUB_ENV
+
+          dv_result=$(forge create --rpc-url http://127.0.0.1:8545 --mnemonic "test test test test test test test test test test test junk" src/double_verify_proof.sol:UltraVerifier)
+          echo "$dv_result"
+          address=$(echo "$dv_result" | grep "Deployed to:" | awk '{print $3}')
+          echo "DV_CONTRACT_ADDRESS=$address" >> $GITHUB_ENV
```

Comment:
`forge create` supports a json output which would be simpler to read the address from.

Reply:
Will replace this implementation with the json output of forge create